geomesa-arrow/geomesa-arrow-datastore/src/main/scala/org/locationtech/geomesa/arrow/data/ArrowFeatureStore.scala

```diff
@@ -0,0 +1,108 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.arrow.data
+
+import java.util.concurrent.atomic.AtomicLong
+
+import org.geotools.data.store.{ContentEntry, ContentFeatureSource, ContentFeatureStore}
+import org.geotools.data.{FeatureReader, FeatureWriter, Query}
+import org.geotools.geometry.jts.ReferencedEnvelope
+import org.locationtech.geomesa.arrow.io.{SimpleFeatureArrowFileReader, SimpleFeatureArrowFileWriter}
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+class ArrowFeatureSource(entry: ContentEntry) extends ContentFeatureSource(entry, Query.ALL) {
+
+  import org.locationtech.geomesa.arrow.allocator
+
+  private [data] val ds = entry.getDataStore.asInstanceOf[ArrowDataStore]
+
+  override def buildFeatureType(): SimpleFeatureType = ds.getSchema()
+
+  override def getBoundsInternal(query: Query): ReferencedEnvelope = null
+
+  override def getCountInternal(query: Query): Int = -1
+
+  override def getReaderInternal(query: Query): FeatureReader[SimpleFeatureType, SimpleFeature] = {
+    val reader = new SimpleFeatureArrowFileReader(ds.createInputStream(), query.getFilter)
+    val features = reader.features
+
+    new FeatureReader[SimpleFeatureType, SimpleFeature] {
+      override def getFeatureType: SimpleFeatureType = reader.sft
+      override def hasNext: Boolean = features.hasNext
+      override def next(): SimpleFeature = features.next()
+      override def close(): Unit = reader.close()
+    }
+  }
+
+  override def canFilter: Boolean = true
+
+  override def canReproject: Boolean = false
+  override def canLimit: Boolean = false
+  override def canSort: Boolean = false
```

Comment:
Yeah, i guess my thought is that it would be bad to sort unless you know that the file is already sorted, and if it's already sorted then you don't need to sort ;)
I'll open a ticket though